examples/capture/main.rs

```diff
@@ -163,43 +162,60 @@ async fn create_png_for_buffer(png_output_path: &str, width: usize, height: usiz
+        Self { width, height, unpadded_bytes_per_row, padded_bytes_per_row }
+    }
+}
+
 fn main() {
     #[cfg(not(target_arch = "wasm32"))]
-    {
-        env_logger::init();
-        futures::executor::block_on(run("red.png"));
-    }
+        {
+            env_logger::init();
+            futures::executor::block_on(run("red.png"));
+        }
     #[cfg(target_arch = "wasm32")]
-    {
-        std::panic::set_hook(Box::new(console_error_panic_hook::hook));
-        console_log::init().expect("could not initialize logger");
-        wasm_bindgen_futures::spawn_local(run("red.png"));
-    }
+        {
+            std::panic::set_hook(Box::new(console_error_panic_hook::hook));
+            console_log::init().expect("could not initialize logger");
+            wasm_bindgen_futures::spawn_local(run("red.png"));
+        }
 }
 
 
 #[cfg(test)]
-mod tests{
+mod tests {
     use super::*;
-    use std::io::Read;
-    use std::fs::remove_file;
+    use std::fs;
 
     #[test]
-    fn ensure_generated_png_matches_expected(){
-        futures::executor::block_on(assert_generated_png_matches_expected(
-            "examples/capture/test_output.png",
-            "examples/capture/screenshot.png"));
+    fn ensure_generated_data_matches_expected() {
+        //fs::read("examples/capture/expected_buffer_data.buffer").unwrap();
+        futures::executor::block_on(assert_generated_data_matches_expected(
+            "examples/capture/expected_buffer_data.buffer"));
     }
 
-    async fn assert_generated_png_matches_expected(test_file_name: &str, expected_file_name: &str) {
-        create_png_for_buffer(test_file_name, 100usize, 200usize).await;
-        let mut test_file = File::open(test_file_name).expect("test output file not found");
-        let mut expected_file = File::open(expected_file_name).expect("expected file not found");
-        let mut test_file_data = Vec::new();
-        test_file.read_to_end(&mut test_file_data).expect("failed to read from test file");
-        let mut expected_file_data = Vec::new();
-        expected_file.read_to_end(&mut expected_file_data).expect("failed to read from expected file");
-        assert_eq!(test_file_data, expected_file_data);
-        remove_file(test_file_name).expect("failed to remove test file");
+    async fn assert_generated_data_matches_expected(file_containing_expected_data: &str) {
+        let (device, output_buffer, _) = create_red_image_with_dimensions(100usize, 200usize).await;
+        let buffer_slice = output_buffer.slice(..);
+        let buffer_future = buffer_slice.map_async(wgpu::MapMode::Read);
+        device.poll(wgpu::Maintain::Wait);
+        buffer_future.await.expect("failed to map buffer slice for capture test");
+        let padded_buffer = buffer_slice.get_mapped_range();
+        let actual_buffer_data = padded_buffer.to_vec();
+        let expected_buffer_data = fs::read(file_containing_expected_data).unwrap();
```

Comment:
do we actually need this? We know that the buffer needs to be all red. Can we just check if all the 4-byte chunks of the data are red? e.g.
```
for chunk in actual_buffer_data.chunks(4) {
  assert_eq!(chunk, &[0xFFu8, 0, 0, 1]);
}
```